src/posthog-surveys.ts

```diff
@@ -214,34 +221,46 @@ export class PostHogSurveys {
             return
         }
 
-        const generateSurveys = phExtensions.generateSurveys
-
         if (!this._decideServerResponse) {
             logger.warn('Decide not loaded yet. Not loading surveys.')
             return
         }
 
-        if (this._surveyEventReceiver == null) {
-            this._surveyEventReceiver = new SurveyEventReceiver(this.instance)
-        }
-
-        if (!generateSurveys) {
-            const loadExternalDependency = phExtensions.loadExternalDependency
-
-            if (loadExternalDependency) {
-                loadExternalDependency(this.instance, 'surveys', (err) => {
-                    if (err) {
-                        logger.error('Could not load surveys script', err)
-                        return
-                    }
-
-                    this._surveyManager = phExtensions.generateSurveys?.(this.instance)
-                })
+        this._isInitializingSurveys = true
+
+        try {
+            const generateSurveys = phExtensions.generateSurveys
+
+            if (!generateSurveys) {
+                const loadExternalDependency = phExtensions.loadExternalDependency
+
+                if (loadExternalDependency) {
+                    loadExternalDependency(this.instance, 'surveys', (err) => {
+                        if (err || !phExtensions.generateSurveys) {
+                            logger.error('Could not load surveys script', err)
+                            this._isInitializingSurveys = false
+                            return
+                        }
+
+                        this._surveyManager = phExtensions.generateSurveys(this.instance)
```

Comment:
you still have to check the return of this function, the signature of `generateSurveys` is `generateSurveys?: (posthog: PostHog) => any | undefined` so it can return undefined and it will be wrongly set to "initialized" but its actually not